extensions/wikia/AdEngine/js/template/floating-rail.js

```diff
@@ -0,0 +1,101 @@
+/*global define*/
+define('ext.wikia.adEngine.template.floating-rail', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.uapContext',
+	'ext.wikia.adEngine.utils.math',
+	'ext.wikia.adEngine.adHelper',
+	'jquery',
+	'wikia.log',
+	'wikia.document',
+	'wikia.window'
+], function (adContext, uapContext, math, adHelper, $, log, doc, win) {
+	'use strict';
+
+	var $medrec = $('#TOP_RIGHT_BOXAD'),
+		$rail = $('#WikiaRail'),
+		$railWrapper = $('#WikiaRailWrapper'),
+		$wikiaMainContent = $('#WikiaMainContent'),
+		$win = $(win),
+		adsInRail = 2,
+		availableSpace,
+		biggestAdSize = 600,
+		globalNavHeight = $('#globalNavigation').height(),
+		floatingSpace,
+		margin = 10,
+		medrecDefaultSize = 250,
+		logGroup = 'ext.wikia.adEngine.template.floating-rail',
+		railWidth = 300,
+		startPosition,
+		stopPosition,
+
+		update = adHelper.throttle(function () {
+			startPosition = parseInt($railWrapper.offset().top, 10) - globalNavHeight - margin;
+			stopPosition = startPosition + floatingSpace;
+
+			// Check if medrec has hidden class for handling tablet mode
+			if ($win.scrollTop() <= startPosition || $medrec.hasClass('hidden')) {
```

Comment:
is it possible to keep styles in styles? What I mean is, can you try to move the styling to scss file and here only add/remove class. For instance in case of tablet mode - you can remove the condition from here and handle this by proper selector.